### DISCUSSION TITLE
Zustand slices pattern with middleware ?

### DISCUSSION THREAD

**Discussion by Twiggeh** [pmndrs/zustand#1480] [Q&A]
I've looked at similar discussions, but I believe the person in question was using a custom middleware ? 

I have tried creating a sliced store, but I did not manage to make it use Middleware correctly, I could also not find a recipe that could help me out. 

[Codesandbox](https://codesandbox.io/s/old-violet-0lf93e?file=/src/App.tsx)

I have here an example store 
```js
const bearStore: StateCreator<
  AllStores,
  [
    ["zustand/subscribeWithSelector", never],
    ["zustand/persist", { fish: number }]
  ],
  [],
  BearStore
> = (set, get) => {
  return {
    addBear: (amount = 1) => set({ bears: get().bears + amount }),
    removeBear: (amount = 1) => set({ bears: get().bears - amount }),
    bears: 1
  };
};
```

that combines into an animalStore

```js
const useAnimalStore = create<AllStores>()(
  persist(
    subscribeWithSelector((...a) => {
      return {
        ...bearStore(...a),
        ...fishStore(...a)
      };
    }),
    {
      name: "my store",
      partialize: (state) => {
        return { fish: [], addBear: 124123 };
      }
    }
  )
);
```

but partialize is not inferring the correct types, and I am getting strange errors on the `...a`

I am not certain whether I am doing something wrong, or whether this is not a supported usecase

**Comment by dai-shi** [comment]
Here's what I've got:

https://tsplay.dev/NrD9VW

The key point seems this:
https://github.com/pmndrs/zustand/blob/b1cea640c82b0cdc2b225502230b1c23e4977c81/docs/guides/typescript.md?plain=1#L405
> Also https://github.com/pmndrs/zustand/issues/980#issuecomment-1162289836 passing actual PersistedState won't work. In those cases, try passing unknown.

**Comment by mooalot** [comment]
I made [zustand-namespaces](https://github.com/mooalot/zustand-namespaces) to allow middleware to be used within slices.
